onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -411,6 +413,77 @@ class DataViewSet(AnonymousUserPublicFormsMixin, ModelViewSet):
 
     queryset = XForm.objects.all()
 
+    def bulk_delete(self, request, *args, **kwargs):
+        """
+        Bulk delete instances
+        """
+        xform = self.__validate_permission_on_bulk_action(request,
+                                                          'change_xform')
+        payload = self.__get_payload(request)
+
+        mongo_query = ParsedInstance.get_base_query(xform.user.username,
+                                                    xform.id_string)
+        postgres_query = {'xform_id': xform.id}
+
+        instances_ids = self.__get_instances_ids(payload, mongo_query)
+
+        if instances_ids is not None:
+            # Narrow down queries with list of ids.
+            postgres_query.update({'id__in': instances_ids})
+            mongo_query.update({'_id': {'$in': instances_ids}})
+
+        # Delete Postgres & Mongo
+        updated_records_count = Instance.objects.filter(**postgres_query).count()
+
+        # Since Django 1.9, `.delete()` returns an dict with number of rows
+        # deleted per object.
+        # FixMe remove `.count()` query and use that dict instance
+        Instance.objects.filter(**postgres_query).delete()
+        ParsedInstance.bulk_delete(mongo_query)
+        return Response({
+            'detail': _('{} submissions have been deleted').format(
+                updated_records_count)
+        }, status.HTTP_200_OK)
+
+    def bulk_validation_status(self, request, *args, **kwargs):
+
+        xform = self.__validate_permission_on_bulk_action(request,
+                                                          'validate_xform')
+        payload = self.__get_payload(request)
+
+        if request.data.get('reset'):
```

Comment:
Would require changes to https://github.com/kobotoolbox/kpi/blob/5544a9a1ab3527e036b5bc485a890e9fcfde614b/kpi/deployment_backends/kobocat_backend.py#L564-L565

Reply:
Actually, there is no need to set `new_validation_status_uid` to `{}` because the `get_validation_status()` returns `{}` if `new_validation_status_uid` is either `{}` or `None`. 
I would avoid adding another `if`.  
It should like this
```
        try:
            new_validation_status_uid = payload['validation_status.uid']
        except KeyError:
            raise ValidationError({
                'payload': _('No `validation_status.uid` provided')
            })

        # Create new validation_status object
        new_validation_status = get_validation_status(
            new_validation_status_uid, xform, request.user.username)
```
What do you think?

Reply:
Looks great